lib/graphql/dig.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module GraphQL
+  module Dig
+    # implemented using the old activesupport #dig instead of the ruby built-in
+    # so we can use some of the magic in Schema::InputObject and Query::Arguments
+    # to handle stringified/symbolized keys.
+    #
+    # @param args [Array<[String, Symbol>] Retrieves the value object corresponding to the each key objects repeatedly
+    # @return [Object]
+    def dig(*args)
```

Comment:
It looks like the first item from this list is used for `self[]`, then the rest are passed on to the resulting object, is that right? Another way to specify that would be: 

```ruby 
def dig(own_key, *rest_keys) 
```

Then, you don't have to do any confusing array modifications. (For example, `self[args.shift]` is a pretty complicated line, performing a read, an array modification, and a key lookup!)

Reply:
good idea, i'll update this.